### ISSUE TITLE
[Game Mode Proposal]: Stepwise

### ISSUE THREAD

**Issue by pianoblook** [Hanabi-Live/hanabi-live#2967]
## Stepwise Mode ##
_(or maybe 'Gearshift Mode'? 'Cyclic Mode'? whatever.)_

### Type 1: Stepwise Numbers ###
- Number clues must be given in sequential order: from 1 to 5, repeating. (e.g. _[1-2-3-4-5-1-2[...]]_)
- If no such number is legally available to clue, <s>a number clue just can't be given that turn.</s> **an empty clue can be used.** *

- The _first_ number clue can be any number, but then the sequence becomes permanently set.

"*" = ideally this means an empty clue is ONLY an option when no *positive* option is available amongst all clueable hands (as a sort of last resort to avoid softlocks) - I don't want this to simply mean "Empty Clues Are On".

### Type 2: Stepwise Colors ###

_(The same as the Number version, except Colors from left to right.)_

### Type 3: Totally Stepwise ###

- Both number and color clues must follow their separate, respective sequences.

**Comment by IceSelkie** [comment]
This sounds similar to the existing alternating clues variant! Maybe the code can be based on that.

Suggestion: Rather than blocking the hint from be given, maybe have it enable empty clues by default: if you must hint 5s, you can tell someone that they don't have 5s!

**Comment by pianoblook** [comment]
> Suggestion: Rather than blocking the hint from be given, maybe have it enable empty clues by default 

Yeah hard to tell which would be more fun so I just went with the more straightforward option. 

I'm usually not a big empty-clue fan, but thinking about it more it's probably worth it to avoid annoyance / potential softlocks (especially possible in 2p, and probably never that uncommon in "Totally" mode). 

Since it's just an edgecase anyway, it's not like it will be that exploitable - and heck, it's probably already pretty difficult as is lol.
